pasqal_cloud/client.py

```diff
@@ -160,10 +199,32 @@ def _send_batch(self, batch_data: Dict[str, Any]) -> Dict[str, Any]:
 
     def _get_batch(self, batch_id: str) -> Dict[str, Any]:
         response: Dict[str, Any] = self._request(
-            "GET", f"{self.endpoints.core}/api/v1/batches/{batch_id}"
+            "GET", f"{self.endpoints.core}/api/v2/batches/{batch_id}"
         )["data"]
         return response
 
+    def _get_batch_jobs(self, batch_id: str) -> list[Dict[str, Any]]:
+        return self._request_with_pagination(
+            "GET",
+            f"{self.endpoints.core}/api/v2/jobs",
+            params={
+                "batch_id": batch_id,
+                "order_by": "ordered_id",
+                "order_by_direction": "ASC",
+            },
+        )
+
+    def _get_job_results(self, job_id: str) -> Dict[str, Any]:
+        results_link = self._request(
+            "GET", f"{self.endpoints.core}/api/v1/jobs/{job_id}/results_link"
+        )["data"]["results_link"]
+        if results_link:
+            try:
+                return self._request("GET", results_link)
+            except HTTPError:
+                pass
```

Comment:
Avoid raising an exception if the pre-signed URL points to a non-existent file (result not uploaded yet)

I noticed in the e2e tests that some jobs move to the DONE status while the results are still not yet uploaded